File: C++/santa_and_gifts.cpp
// Santa and Gifts

#include <bits/stdc++.h>
using namespace std;

int fn(vector<int> &a, int n, int c)
{
    sort(a.begin(), a.end());
    vector<int>::iterator it; 
    while(a.size() != c)
    {
        int x = a[0] + a[1], i;
        it = a.begin();
        a.erase(it); 
        it = a.begin(); 
        a.erase(it);
        a.push_back(x);
        
        for(i = a.size()-2; a[i]>x; i--)
            a[i+1] = a[i];
        
        a[i+1] = x;
    }
    return a[c-1];
}

int main() 
{
	int t, n, c, x;
	cin>>t;
	while(t--)
	{
	    cin>>n;
	    vector<int> a;
	    for(int i=0; i<n; i++)  
	    {
	        cin>>x;
	        a.push_back(x);
	    }
	    cin>>c;
	    if(c>n)  cout<<-1<<endl;
	    else  cout<<fn(a, n, c)<<endl;
	}
	return 0;
}

File: Python/santa_and_gifts.py
# Santa and Gifts

def fn(a, n, c):
    a.sort()
    while len(a) != c:
        x = a[0] + a[1]
        del(a[0])
        del(a[1])
        a.append(x)
        a.sort()
    return a[c-1]


if __name__ == '__main__': 
    t = int(input())
	for i in range(t):
	    n = int(input())
	    a = []
	    for j in range(n):
	        x = int(input())
	        a.append(x)
	    c = int(input())
	    if c>n:  c
	        print(-1)
	    else:
	        print(fn(a, n, c))
